### ISSUE TITLE
Login Screen Loses Provided User Email on Device Orientation in Android

### ISSUE THREAD

**Issue by dinobu** [AzureAD/azure-activedirectory-library-for-dotnet#1622]
I have Xamarin.Forms application that authenticates user against Azure AAD. That all works fine but on Android, device orientation looses user email on the Microsoft authentication screen.

Here, my device is in Portrait mode and I have entered user email:

![Screenshot_20190630-093043](https://user-images.githubusercontent.com/40807298/60401262-69ab9280-9b33-11e9-80d1-65cff537fef0.png)

If I enter user email and click on Next, I land on screen prompting me to enter password. If I now rotate device on Android, it will return me back to the screen above but notice, the screen has lost my previously entered user email, it is blank now

![Screenshot_20190630-093002](https://user-images.githubusercontent.com/40807298/60401156-387e9280-9b32-11e9-8093-59a2e6437ac0.png)

This is not happening on IOS devices, only Android.  Device rotation should not return me back to screen prompting for user email not should it loose that information.  Similarly, If I have entered password half way through and device is rotated, I should be just able to continue entering the rest of my password.

How do I prevent the rotation on this Sign in screen from loosing already entered user name and re-prompting for it? I dont want to disable rotation.

My MainActivity has already ConfigChages.Orientation attribute like below:

```
[Activity(Name = "my.mainactivity"
, Label = "MyApp"
, Icon = "@drawable/icon"
, ConfigurationChanges = ConfigChanges.ScreenSize 
    | ConfigChanges.SmallestScreenSize 
    | ConfigChanges.ScreenLayout 
    | ConfigChanges.Orientation)]
public class MainActivity : Xamarin.Forms.Platform.Android.FormsAppCompatActivity
{
   ...
}
```

, but this is not resolving the issue and I think the reason is that AcquireTokenAsync is probably using WebView which refreshes itself on device rotation on Android.  How do I prevent this?  I dont think your API provides a way around this which is very annoying issue on mobile devices.

**Comment by bgavrilMS** [comment]
No new features will be implemented on ADAL. The team's efforts are on improving [MSAL](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet), the next-gen auth library. MSAL's wiki contains a [migration guide](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Adal-to-Msal) from ADAL.

Only regressions, high severity issues and security issues will be fixed on ADAL. Other issues are likely to have already been fixed in MSAL. If not, please log them and the team will prioritise them.

**Comment by henrik-me** [comment]
@dinobu : We may have to address the same type of issue in MSAL, thus pls. open the issue there. We believe it's relatively simple to move from ADAL to MSAL however should you need assistance we are happy to help.

**Comment by jennyf19** [comment]
@dinobu I am able to reproduce the issue in MSAL. would you mind opening an [issue there](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/new?template=feature_request.md)? thanks.

cc: @henrik-me

**Comment by jennyf19** [comment]
I'll link this [issue](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1244), as, I think, they are related.

**Comment by dinobu** [comment]
> 
> 
> I'll link this [issue](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1244), as, I think, they are related.

Thank you @jennyf19

**Comment by henrik-me** [comment]
@dinobu : We would like to make this experience better. What is preventing you from moving to MSAL?

**Comment by dinobu** [comment]
> @dinobu : We would like to make this experience better. What is preventing you from moving to MSAL?

No plans to move to MSAL but we are using Xamarin.Forms app (all 3 platforms) to authenticate with AzureAD, if you can provide me good guide to move to MSAL, I will take a look.  But again, this is not planned to do, it is just my personal interest that I would like to take a look.  We still need the ADAL fix.

**Comment by jennyf19** [comment]
@dinobu [Here's an overview of differences between ADAL and MSAL](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Adal-to-Msal) and also [benefits of Microsoft identity platform v2](https://docs.microsoft.com/en-us/azure/active-directory/develop/azure-ad-endpoint-comparison).

**Comment by dinobu** [comment]
> @dinobu [Here's an overview of differences between ADAL and MSAL](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Adal-to-Msal) and also [benefits of Microsoft identity platform v2](https://docs.microsoft.com/en-us/azure/active-directory/develop/azure-ad-endpoint-comparison).

Thanks, I have already read both of these but will still take another look.  Much appreciated.

**Comment by jennyf19** [comment]
@dinobu Great...if you have any questions, let us know.

**Comment by dinobu** [comment]
Any movement on the hot fix for ADAL to resolve this issue?

**Comment by bgavrilMS** [comment]
Sorry @dinobu , but this is not a regression and we will not fix this in ADAL. I am not sure when we'll be able to take a look at it in MSAL, but feel free to propose smth if you are so inclined. We accept contributions.

**Comment by jennyf19** [comment]
@dinobu this fix will be in the 5.1.1 release, which will be out this week. Fix is also in MSAL and will be in next MSAL release.

**Comment by dinobu** [comment]
> @dinobu this fix will be in the 5.1.1 release, which will be out this week. Fix is also in MSAL and will be in next MSAL release.

Thanks again, will grab ADAL 5.1.1 once released.  Much appreciated @jennyf19

**Comment by jennyf19** [comment]
Included in [5.1.1 release](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/releases/tag/5.1.1)
